src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -457,6 +457,11 @@ private static object CalculatePageName(ActionContext actionContext, string page
                     throw new InvalidOperationException(Resources.FormatUrlHelper_RelativePagePathIsNotSupported(pageName));
                 }
 
+                if(pageName.StartsWith("./"))
```

Comment:
Should we do back-slashes too? `ViewEnginePath` allows `..\` for path navigation.